src/main/kotlin/org/rust/lang/core/resolve/ResolveEngine.kt

```diff
@@ -212,6 +209,26 @@ object ResolveEngine {
         return module.project.getPsiFor(module.cargoWorkspace?.findCrateByName(name)?.crateRoot)?.rustMod
     }
 
+    fun resolveLabel(label: RsLabel): RsLabelDecl? =
+        label.ancestors
+            .takeWhile { it !is RsLambdaExpr && it !is RsFunction }
+            .mapNotNull { it as? RsLabeledExpression }
+            .map { it.labelDecl }
+            .find { it.name ==  label.quoteIdentifier.text}
+
+    fun resolveLifetime(lifetimeRef: RsLifetime): RsLifetimeDecl? {
+        if (lifetimeRef.isPredefined) return null
+        return lifetimeRef.ancestors
+            .mapNotNull {
+                when (it) {
+                    is RsTypeAlias -> if (it.parent is RsImplItem || it.parent is RsTraitItem) null else it.typeParameterList
```

Comment:
> and annotate lifetimes in associated types as forbidden

We already do that.

I just thought we could annotate an undeclared lifetime earlier, without waiting for the user to remove an invalid lifetime declaration from the type alias in cases like this:

```rust
trait Foo {
    type Bar<'a> = &'a str;
}
```

But ok, it's a very minor improvement and maybe doesn't worth carrying this logic.